R/build-search-docs.R

```diff
@@ -29,17 +29,20 @@ build_docsearch_json <- function(pkg = ".") {
 build_sitemap <- function(pkg = ".") {
   pkg <- as_pkgdown(pkg)
 
-  url <- pkg$meta$url
+  url <- paste0(pkg$meta$url, "/")
   if (is.null(url)) {
     return()
   }
 
+  if (pkg$development$in_dev && pkg$bs_version > 3) {
+    url <- paste0(url, pkg$prefix)
+  }
+
   urls <- paste0(
-    url, "/",
-    c(
-      path("index.html"),
-      path("reference", unique(pkg$topics$file_out)),
-      path(pkg$vignettes$file_out)
+    url,
```

Comment:
maybe this change is a bad idea, as it might mean Algolia indexes more pages than what BS3 users were used to.
but then, the sitemap might be used for other things one day?

Reply:
Yeah, I think it makes more sense to index everything.

I think the biggest downside of this is that (IIRC) the currently the pkgdown workflow doesn't clean up old files, so this would include old topics/articles in the index. I don't know if we should fix that here or elsewhere.

Reply:
I made a note in the corresponding issue.